Model/CustomItemExportSchedulerModel.php

```
@@ -0,0 +1,294 @@
+
+        $this->em->persist($customItemExportScheduler);
+        $this->em->flush();
+
+        return $customItemExportScheduler;
+    }
+
+    public function processDataAndGetExportFilePath(CustomItemExportScheduler $customItemExportScheduler): string
+    {
+        $scheduledDateTime = $customItemExportScheduler->getScheduledDateTime();
+        $fileName          = 'custom_items_export_'.$scheduledDateTime->format(self::EXPORT_FILE_NAME_DATE_FORMAT).'.csv';
+
+        $filePath = $this->exportHelper->getValidExportFileName($fileName, 'custom_item_export_dir');
+
+        $this->filePath = $filePath;
+
+        $this->processCustomItemData($customItemExportScheduler);
+
+        return $this->exportHelper->zipFile($this->filePath, 'custom_item_exports.csv');
+    }
+
+    private function processCustomItemData(CustomItemExportScheduler $customItemExportScheduler): void
+    {
+        $customObjectRepo = $this->em->getRepository(CustomObject::class);
+
+        $customObject = $customObjectRepo->find($customItemExportScheduler->getCustomObject());
+
+        $customFields = $customObject->getCustomFields()->toArray();
+
+        $this->addExportFileHeaderToCsvFile($customFields);
+
+        $this->addCustomItemsToCsvFile($customFields, $customObject);
+    }
+
+    /**
+     * @param array<CustomField> $customFields
+     *
+     * @return array<string>
+     */
+    private function getCSVHeader(array $customFields): array
+    {
+        $header = ['customItemId', 'customItemName'];
+
+        foreach ($customFields as $customField) {
+            $header[] = $customField->getId();
+        }
+
+        $header[] = 'linkedContactIds';
+
+        return $header;
+    }
+
+    /**
+     * @param array<CustomField> $customFields
+     */
+    private function addCustomItemsToCsvFile(array $customFields, CustomObject $customObject): void
+    {
+        $offset = 0;
+        $result = true;
+
+        $handler = @fopen($this->filePath, 'ab+');
```

Do we really want to hide the error here?
I would suggest we use the finder component over using php for better flexibility.